arbnode/node.go

```diff
@@ -704,6 +700,11 @@ func createNodeImpl(
 		if err != nil {
 			return nil, err
 		}
+
+		// Check if staker and batch poster are using the same address
+		if stakerAddr != (common.Address{}) && !strings.EqualFold(config.Staker.Strategy, "watchtower") && stakerAddr == batchPoster.dataPoster.Sender() {
+			return nil, fmt.Errorf("staker and batch poster the same data poster address: %v", stakerAddr)
```

Comment:
This is an error that will probably be hit by users, so let's make it a bit clearer. Something like: "staker and batch poster are using the same address which is not allowed: %v"

Most users probably won't know what the data poster is